File: ios/Khrysalis Template/com/lightningkite/khrysalistemplate/vg/DateButtonDemoVG.shared.swift
//Package: com.lightningkite.khrysalistemplate.vg
//Converted using Khrysalis2

import Foundation
import Khrysalis
import RxSwift
import RxRelay
import KhrysalisMaps



public class DateButtonDemoVG: ViewGenerator {
    
    
    override public var title: String {
        get {
            return "Date Button Demo"
        }
    }
    public var date: MutableObservableProperty<Date>
    
    override public func generate(dependency: ViewDependency) -> View {
        var xml = DateButtonDemoXml()
        var view = xml.setup(dependency)
        xml.text.bindString(date.map{ (it) in 
            it.format(ClockPartSize.Medium, ClockPartSize.Medium)
        })
        xml.dateButton.bind(date)
        xml.timeButton.bind(date)
        return view
    }
    override public func generate(_ dependency: ViewDependency) -> View {
        return generate(dependency: dependency)
    }
    
    override public init() {
        let date: MutableObservableProperty<Date> = StandardObservableProperty(Date())
        self.date = date
        super.init()
    }
}
 

File: android/src/main/java/com/lightningkite/khrysalistemplate/vg/DateButtonDemoVG.shared.kt
package com.lightningkite.khrysalistemplate.vg

import android.view.View
import com.lightningkite.khrysalis.time.*
import com.lightningkite.khrysalis.observables.binding.bind
import com.lightningkite.khrysalis.observables.binding.bindString
import com.lightningkite.khrysalis.observables.MutableObservableProperty
import com.lightningkite.khrysalis.observables.StandardObservableProperty
import com.lightningkite.khrysalis.observables.map
import com.lightningkite.khrysalis.views.ViewDependency
import com.lightningkite.khrysalis.views.ViewGenerator
import com.lightningkite.khrysalistemplate.layouts.DateButtonDemoXml
import java.util.*

class DateButtonDemoVG() : ViewGenerator() {
    override val title: String get() = "Date Button Demo"

    val date: MutableObservableProperty<Date> = StandardObservableProperty(Date())

    override fun generate(dependency: ViewDependency): View {
        val xml = DateButtonDemoXml()
        val view = xml.setup(dependency)

        xml.text.bindString(date.map { it -> it.format(ClockPartSize.Medium, ClockPartSize.Medium) })
        xml.dateButton.bind(date)
        xml.timeButton.bind(date)

        return view
    }
}
